### PR TITLE
Fixes #21894 Remove redundant bean-validation implementations from GlassFish

### LINKED ISSUE #21894: Remove redundant bean-validation implementations from GlassFish
## Environment Details
* GlassFish Version (and build number): 5.0
* JDK version: Any 1.8 version
* OS: Any version
* Database: N.A

----------

## Problem Description
GlassFish 5.0 builds seem to be bundling two instances of bean validator. One by name hibernate-validator.jar and another by name bean-validator.jar

bean-validator.jar seems to be a uber jar that has the content of hibernate-validator.jar and all of its dependencies.e.g: javax.validation APIs, jboss logging implementation and hibernate validator implementation classes.

It looks like the reason why bean-validator.jar was introduced seems to be to add osgi headers in the MANIFEST.MF which hibernate-validator.jar was not doing before.
(Refer the removed hk2/external modules meant to create the uber jar)
https://github.com/javaee/hk2/commit/22cb22a13cb42db88babb1f0487955d5d7e51b14

Now, atleast from hibernate 5.2.4 (BV 1.1 implementation), the hibernate-validator.jar has OSGi headers present.

So, we should remove one of them (preferably bean-validator.jar) so that it is easier to introduce the next version of bean validation implementation (i.e., javax.validation 2.0 and its implementation).

### PR DESCRIPTION
**Pull request by h2002044**
Fixes #21894
Please refer issue https://github.com/javaee/glassfish/issues/21894 for all description.
(Same changes as that of PR https://github.com/javaee/glassfish/pull/21911, but with updated copyright headers for the new files and submitting as a single commit).

### CODE REVIEW

**Comment by h2002044** [comment]
@glassfishrobot Run CI tests please

**Review by mtau11 — DISMISSED** [review]

**Review by ankathur — DISMISSED** [review]

**Comment by h2002044** [comment]
@glassfishrobot Run CI tests please

File: nucleus/admin/cli/pom.xml [hunk 1]
```diff
@@ -120,6 +120,9 @@
                         <manifest>
                             <mainClass>com.sun.enterprise.admin.cli.AdminMain</mainClass>
                         </manifest>
+                        <manifestEntries>
```
Comment (yaminikb) [inline nucleus/admin/cli/pom.xml]: Why is this needed?

File: nucleus/packager/external/bean-validator-cdi/pom.xml [hunk 2]
```diff
@@ -0,0 +1,198 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+
+    Copyright (c) 1997-2017 Oracle and/or its affiliates. All rights reserved.
```
Comment (yaminikb) [inline nucleus/packager/external/bean-validator-cdi/pom.xml]: Copyright year should be fixed. Since its a new file, it should be 2017.

File: nucleus/packager/nucleus-hk2/pom.xml [hunk 3]
```diff
@@ -94,18 +94,19 @@
             <groupId>org.glassfish.external</groupId>
             <artifactId>asm-all</artifactId>
         </dependency>
-        <dependency>
+<!--        <dependency>
```
Comment (yaminikb) [inline nucleus/packager/nucleus-hk2/pom.xml]: Is there a reason why this is left commented instead of deleting?

File: nucleus/pom.xml [hunk 4]
```diff
@@ -124,6 +124,8 @@
 
         <glassfish-corba.version>4.0.1</glassfish-corba.version>
         <stax-api.version>1.0-2</stax-api.version>
+        <slf4j.version>1.7.21</slf4j.version>
```
Comment (yaminikb) [inline nucleus/pom.xml]: SLF4J latest is 1.7.25, shouldn't we choose the latest?

File: nucleus/pom.xml [hunk 5]
```diff
@@ -147,7 +149,8 @@
         <jackson.version>2.8.4</jackson.version>
         <jettison.version>1.3.3</jettison.version>
         <jax-rs-api.spec.version>2.1</jax-rs-api.spec.version>
-        <jax-rs-api.impl.version>2.1-m09</jax-rs-api.impl.version>
+        <jax-rs-api.impl.version>2.1-m08</jax-rs-api.impl.version>
```
Comment (yaminikb) [inline nucleus/pom.xml]: Looks like this is unintentional.

File: nucleus/pom.xml [hunk 6]
```diff
@@ -147,7 +149,8 @@
...
+        <jtype.version>0.1.0</jtype.version>
```
Comment (yaminikb) [inline nucleus/pom.xml]: Again, jtype 0.1.3 is latest

**Review by yaminikb — CHANGES_REQUESTED** [review]

File: nucleus/admin/cli/pom.xml (hunk 1 above)
Comment (h2002044) [inline nucleus/admin/cli/pom.xml]: There are duplicate jars for bean-validator and bean-validator-cdi.
We are removing hibernate-validator and hibernate-validator-cdi jars and continue to bundle bean-validator and bean-validator-cdi (custom made jars that has all dependencies).
admin-cli module and its downstream dependencies make use of hibernate validator. Today (before this change), the maven plugin adds all dependencies of "hk2" (hibernate validator being one of them) in its classpath. Since we are removing hibernate validator, we are explicitly adding bean-validator in the classpath. (bean-validator now being hosted as part of GlassFish modules, it is not auto detected during class-path entry detection and hence we are adding it explicitly).

File: nucleus/packager/external/bean-validator-cdi/pom.xml (hunk 2 above)
Comment (h2002044) [inline nucleus/packager/external/bean-validator-cdi/pom.xml]: This is a packaging module copied from hk2 as HK2 team does not support this any more and want GlassFish modules to take care of it. So, this is not a new file.

File: nucleus/packager/nucleus-hk2/pom.xml (hunk 3 above)
Comment (h2002044) [inline nucleus/packager/nucleus-hk2/pom.xml]: we can remove it, I shall remove it when I upgrade hibernate.

File: nucleus/pom.xml (hunk 4 above)
Comment (h2002044) [inline nucleus/pom.xml]: These were the dependencies used by bean-validator packaging module in hk2. I do not want to upgrade those and see different set of issues. We can upgrade to appropriate version of sl4j depending upon the version required by bean-validator 2.0 whenever we introduce bean validator 2.0 support.

File: nucleus/pom.xml (hunk 6 above)
Comment (h2002044) [inline nucleus/pom.xml]: Same comment as that of sl4j

File: nucleus/pom.xml (hunk 5 above)
Comment (h2002044) [inline nucleus/pom.xml]: Yes, looks like a merge issue. I shall fix it.

**Comment by h2002044** [comment]
@glassfishrobot Run CI tests please

**Comment by h2002044** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — APPROVED** [review]
Looks good
